### ISSUE TITLE
SourceLocation should be serialized as object

### ISSUE THREAD

**Issue by ktosiek** [graphql-python/graphql-core-next#61]
`SourceLocation`, as a NamedTuple, is serialized by json.dumps() to an array instead of an object. Graphql.js [keeps `SourceLocation` as and object](https://github.com/graphql/graphql-js/blob/57232347e71b7b23fbcd4c53c8afca73f89e9cc7/src/language/location.js#L8), so it just works for them.

Reproduction:
```python
from graphql import SourceLocation, GraphQLError, format_error, Source
import json

print(json.dumps(format_error(GraphQLError(message="test", source=Source('{ test }'), positions=[2]))))
```

Expected result: `{"message": "test", "locations": [`**`{"line": 1, "column": 3}`**`], "path": null}`
Actual result: `{"message": "test", "locations": [`**`[1, 3]`**`], "path": null}`

Should `SourceLocation` be changed to something that serializes correctly? Or would it make more sense to convert it to dict in `format_errors`?

**Comment by Cito** [comment]
Good point. Yes, I think it makes totally sense to return locations as a list of dicts in `format_error` - that's why we have this function. Feel free to send a PR, or otherwise I can change this.

**Comment by Cito** [comment]
This is now included in [v3.0.0b1](https://github.com/graphql-python/graphql-core-next/releases/tag/v3.0.0b1).
